### PR TITLE
feat(withDirty): combined the withDirty examples with the input examples

### PR DESCRIPTION
**Pull request by dmgauthier**
### Proposed Changes

Combined the Dirty management HOC components examples with the input examples 

### Potential Breaking Changes

Only affect the demo

### Acceptance Criteria

-   [ ] The proposed changes are covered by unit tests
-   [ ] The potential breaking changes are clearly identified
-   [ ] [README.md](https://github.com/coveo/react-vapor/blob/master/README.md) is adjusted to reflect the proposed changes (if relevant)

### CODE REVIEW

File: packages/react-vapor/src/hoc/withDirty/withDirty.tsx [hunk 1]
```diff
@@ -22,7 +22,7 @@ export interface IWithDirtyDispatchProps {
 export interface IWithDirtyProps extends Partial<IWithDirtyStateProps>, Partial<IWithDirtyDispatchProps> {}
 
 export const withDirty = <T, R = any>(config: IWithDirty) => (
-    Component: React.ComponentClass<IWithDirtyProps & T, R>
+    Component: React.ComponentClass<IWithDirtyProps & T, R> | React.FunctionComponent<IWithDirtyProps & T>
```
Comment (gdostie) [inline packages/react-vapor/src/hoc/withDirty/withDirty.tsx]:
```suggestion
    Component: React.ComponentType<IWithDirtyProps & T>
```

File: packages/react-vapor/src/hoc/withEditing/withEditing.tsx [hunk 2]
```diff
@@ -11,7 +11,7 @@ export interface IWithEditing {
 }
 
 export const withEditing = <T, R = any>(config: IWithEditing) => (
-    Component: React.ComponentClass<T, R>
+    Component: React.ComponentClass<T, R> | React.FunctionComponent<T>
```
Comment (gdostie) [inline packages/react-vapor/src/hoc/withEditing/withEditing.tsx]:
```suggestion
    Component: React.ComponentType<T>
```

File: packages/react-vapor/src/hoc/withDirty/withDirty.tsx (hunk 1 above)
Comment (dmgauthier) [reply packages/react-vapor/src/hoc/withDirty/withDirty.tsx]: oh YEAH, thx !!!

File: packages/react-vapor/src/hoc/withEditing/withEditing.tsx (hunk 2 above)
Comment (dmgauthier) [reply packages/react-vapor/src/hoc/withEditing/withEditing.tsx]: :+1: :hugs:

**Pull request merged**